dm/pkg/schema/tracker.go

```diff
@@ -72,14 +72,14 @@ type Tracker struct {
 type downstreamTracker struct {
 	downstreamConn *dbconn.DBConn                  // downstream connection
 	stmtParser     *parser.Parser                  // statement parser
-	tableInfos     map[string]*downstreamTableInfo // downstream table infos
+	tableInfos     map[string]*DownstreamTableInfo // downstream table infos
 }
 
-// downstreamTableInfo contains tableinfo and index cache.
-type downstreamTableInfo struct {
-	tableInfo        *model.TableInfo   // tableInfo which comes from parse create statement syntaxtree
-	indexCache       *model.IndexInfo   // index cache include pk/uk(not null)
-	availableUKCache []*model.IndexInfo // index cache include uks(data not null)
+// DownstreamTableInfo contains tableinfo and index cache.
+type DownstreamTableInfo struct {
+	TableInfo        *model.TableInfo   // tableInfo which comes from parse create statement syntaxtree
+	IndexCache       *model.IndexInfo   // index cache include pk/uk(not null)
```

Comment:
how about rename `IndexCache` to `UkIndexInfo` and `availableUKCache` to `UKIndexInfoList`? this may make code easier to read

Reply:
plz also change the comment and func name

Reply:
`IndexCache` is pk/uk not null and `AvailableUKCache` is uks. i will optimize the description.

Reply:
yes,  but i think we treat pk and uk in the same way(to gen where), so in our case maybe pk is another type of uk too and maybe use `UkIndexInfo` is enough?

just a small suggestion, i can accept current name but with more clearly description

Reply:
How about rename `IndexCache` to  `AbsoluteUKIndexInfo`  and `availableUKCache` to `AvailableUKIndexList`, this can distinguish `pickedDownstreamIndexInfo` which comes from pk/uk not null/uk value not null in `dml.go`.
The logic will like that if there is no `AbsoluteUKIndexInfo` (pk/uk not null), we will pick a index from `AvailableUKIndexList`( uk value not null) as `dml.go`'s `pickedDownstreamIndexInfo` which is used to `genwhere`.

Reply:
LGTM